OpenSim/Simulation/MarkersReference.cpp

```diff
@@ -153,4 +148,37 @@ void MarkersReference::setMarkerWeightSet(const Set<MarkerWeight>& markerWeights
     upd_marker_weights() = markerWeights;
 }
 
+void MarkersReference::setDefaultWeight(double weight)
+{
+    set_default_weight(weight); 
+}
+
+void MarkersReference::updateInternalWeights() const
+{
+    // if weights are not being changed, do not rebuild list of weights.
+    if (isObjectUpToDateWithProperties())
+        return;
+
+    //start by assigning default value to each
+    _weights.assign(getNumRefs(), get_default_weight());
+
+    // if the set of marker weights has not been changed, nothing
+    // else to be done.
+    if (get_marker_weights().isObjectUpToDateWithProperties())
+        return;
+
+    int wix = -1;
+    int ix = 0;
+    // Build flat lists of marker weights in the same order as the marker names
+    for (const std::string &name : _markerNames) {
+        wix = get_marker_weights().getIndex(name, wix);
+        //Assign user weighting for markers that are user listed in the input set
+        if (wix >= 0)
+            _weights[ix++] = get_marker_weights()[wix].getWeight();
```

Comment:
Good point. I added details about the markers being assigned a default weight to begin with.